### ISSUE TITLE
Reg/Operation - Status

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#2222]
#### Describe the task

The `Operation` model currently has a `status` column, which we still want, but it will only be used to indicate whether an operation has completed registration or not.

We will need to add a new column on the `OperationDesignatedOperationTimeline` model to indicate the status of the operation in relation to its connection to the operator.

For `Operation.Status`, the valid options will be:
- Draft
- Registered
- Not Started

Valid for now:
 * Approved
 * Declined
 * Changes Requested
 * Pending

* will be keeping these statuses in place for now to maintain consistency with Reg 1. Once Reg 2 has fully launched, will need to write a data migration script that converts the statuses of operations in prod appropriately

For `OperationDesignatedOperatorTimeline.Status`, the valid options will be:
- Active
- Transferred
- Closed
- Temporarily Shutdown

#### Acceptance Criteria

- [x] write django migration that adds `status` column to `OperationDesignatedOperatorTimeline` model, with TextChoices listed above
- [x] ensure the TextChoices for `status` column on `Operation` model are only those listed above
- [x] for users viewing their list of operations in the data grid, the data being queried should come from OperationDesignatedOperatorTimeline so that the correct status can be fetched. See the facilities grid for how to set this up--the filtering schema is a little different for fk fields
- [ ] update mock data fixtures with correct "status" values (Note from BC: can't do this for all of them or else we break reg1 e2e tests)

#### Additional context

- 
- Or here

**Comment by Sepehr-Sobhani** [comment]
What should we do with the existing statuses for the Operation data model? Some of them, such as `CHANGES_REQUESTED`, `DECLINED`, and `APPROVED`, are from Reg1.

**Comment by patriciarussellCAS** [comment]
@andrea-williams to discuss when you're back :)

**Comment by andrea-williams** [comment]
> What should we do with the existing statuses for the Operation data model? Some of them, such as `CHANGES_REQUESTED`, `DECLINED`, and `APPROVED`, are from Reg1.

updated ticket to keep these statuses from Reg 1 in place for the time being

**Comment by zoeyli-46** [comment]
Do we need to take out "Closed" and "Temporarily Shutdown" for now, since reporting a closure/shutdown is not in MVP? or should we keep it in for now since we may implement it in the future?

**Comment by andrea-williams** [comment]
@zoeyli-46  to simmer on it

**Comment by patriciarussellCAS** [comment]
Question re: 

For OperationDesignatedOperatorTimeline.Status, the valid options will be:
Active
Transferred
Closed
Temporarily Shutdown

Do we have this set up for facilities as well? Or just Operations?

**Comment by Sepehr-Sobhani** [comment]
Yes, we have this for `FacilityDesignatedOperationTimeline` table too.

**Comment by patriciarussellCAS** [comment]
@Sepehr-Sobhani what should I be looking at for this review? in Dev, i can see all Operations are listed as "active" in the status column. When I tried to register Operation 16, I have no way to seeing the registration status. Where should I see this?

**Comment by Sepehr-Sobhani** [comment]
In this PR, we updated the `Operation` data model by renaming some statuses. The `OperationDesignatedOperatorTimeline` table now populates the Operations data grid. Internal users will only see Operations with the "Registered" status, while Operations with "Draft" or "Not-started" statuses are visible exclusively to external users.

Given this change, displaying the status column based on `OperationDesignatedOperatorTimeline` for external users may no longer be meaningful. External users only see "Active" Operations, while internal users see additional statuses like "Transferred," "Closed," etc. Perhaps we could use the status from the `OperationDesignatedOperatorTimeline` model for internal users and the status from the `Operation` model for external users. This approach would allow us to display the appropriate status for both user types.
